src/tools/crawler/metrics.rs

```diff
@@ -61,6 +65,56 @@ pub fn new_network_summary(crawler: &Crawler, graph: &Graph<SocketAddr>) -> Netw
         }
     }
 
+    let mut node_network_types = Vec::with_capacity(num_good_nodes);
+    for node in &good_nodes {
+        let mut port_matches = false;
+        let mut agent_matches = false;
+        let mut height_matches = false;
+        let mut blocker = false;
+
+        if node.port() == ZCASH_P2P_PORT {
+            port_matches = true;
+        }
+
+        let agent = if let Some(agent) = &nodes[node].user_agent {
+            agent.0.clone()
+        } else {
+            "".to_string()
+        };
+        let zcash_re = Regex::new(r"^/MagicBean:(\d)\.(\d)\.(\d)/$").unwrap();
+        let zebra_re = Regex::new(r"^/Zebra:(\d)\.(\d)\.(\d)").unwrap();
+
+        let cap_zc = zcash_re.captures(agent.as_str());
+        if let Some(cap) = cap_zc {
+            let major = cap.get(1).unwrap().as_str().parse::<u32>().unwrap();
+            if major < 6 {
+                // Accept all zcash versions < 6 (6 is Flux)
+                agent_matches = true;
+            } else if major == 6 {
+                blocker = true; // Block all zcash versions 6 (Flux) even if they are on the right port
```

Comment:
a general comment regarding this part and the line 105:

if the must-condition is not met (`height is not okay` || `major==6`), we can just call
```rust
node_network_types.push(NetworkType::Unknown);
continue;
```

so the logic on line 111 gets easier to read.
also we will btw remove  those extra variables `height_matches` and `blocker`

Reply:
> Probably depends on one's preferences 

that's so true!
that's why I asked for some extra comments which provide more context while keeping things simple 😃